packages/amplify-go-function-runtime-provider/src/runtime.ts

```diff
@@ -161,8 +165,14 @@ export const packageResource = async (request: PackageRequest, context: any): Pr
 };
 
 const winZip = async (src: string, dest: string, print: any) => {
-  // get lambda zip tool
-  await execa(executableName, ['get', '-u', 'github.com/aws/aws-lambda-go/cmd/build-lambda-zip']);
+  // get lambda zip tool with the fix of https://go.dev/doc/go-get-install-deprecation
+  try {
+  // for golang version above > 1.17
+    await execa(executableName, ['install', 'github.com/aws/aws-lambda-go/cmd/build-lambda-zip']);
+  } catch (error: unknown) {
+  // for golang version below < 1.17
+    await execa(executableName, ['get', '-u', 'github.com/aws/aws-lambda-go/cmd/build-lambda-zip']);
+  }
```

Comment:
Is there any way to get the go version so that only 1 of these is run? It looks like [this StackOverflow post](https://stackoverflow.com/questions/59820549/how-to-extract-only-version-from-go-version-command-using-sed-or-other-bash-co) has directions for piping out just the number. Then you can use an if/else statement instead of a try/catch.

Reply:
@danielleadams Please check it again.